cis194/week8/stevemao/Party.hs

```diff
@@ -0,0 +1,49 @@
+{-# OPTIONS_GHC -Wall #-}
+module Party where
+import Employee
+import Data.Tree
+import Data.List
+import Text.Read
+
+glCons :: Employee -> GuestList -> GuestList
+glCons e@(Emp _ efun) (GL el glfun) = GL (el ++ [e]) (glfun + efun)
+
+instance Semigroup GuestList where
+  GL el1 f1 <> GL el2 f2 = GL (el1 <> el2) (f1 + f2)
+
+instance Monoid GuestList where
+  mempty = GL [] 0
+
+moreFun :: GuestList -> GuestList -> GuestList
+moreFun gl1 gl2
+  | compare gl1 gl2 == GT = gl1
+  | otherwise = gl2
```

Comment:
We need something like `Ord a => a -> a -> a`. Do we have a function that we can use under `Ord` typeclass?